safe_transaction_service/history/tests/test_tx_processor.py

```diff
@@ -101,6 +103,17 @@ def test_tx_processor_with_factory(self):
         self.assertEqual(safe_status.nonce, 1)
         self.assertEqual(safe_status.threshold, threshold)
 
+        safe_contract = SafeContract.objects.get(address=safe_address)
+        safe_contract_delegate = SafeContractDelegateFactory(
+            delegator=owner, safe_contract=safe_contract
```

Comment:
I think you can use `safe_contract=safe_address` or `safe_contract_id=safe_address` so you don't have to retrieve the object to call the factory

Reply:
Factory should be creating the Safe if you set `safe_contract__address=address`, but it shouldn't be the case. Also I'm not an expert on `factory_boy`, so it looks good this way then 😃

Reply:
You are right, I had tried `safe_contract=safe_address` (causes an error because it requires an instance) but not `safe_contract_id=safe_address`. This way it works as you indicated in the first comment (Lesson learned).

Reply:
Lesson learned also for me, thanks!